src/evict/evict_lru.c

```diff
@@ -841,10 +780,15 @@ __wt_evict_file_exclusive_on(WT_SESSION_IMPL *session)
 	 * this point.
 	 */
 	F_SET(btree, WT_BTREE_NO_EVICTION);
+	F_SET(cache, WT_CACHE_CLEAR_WALKS);
 	WT_FULL_BARRIER();
 
 	/* Clear any existing LRU eviction walk for the file. */
-	WT_ERR(__evict_request_clear_walk(session));
+	__wt_spin_lock(session, &cache->evict_pass_lock);
+	ret = __evict_clear_walk(session);
+	__wt_spin_unlock(session, &cache->evict_pass_lock);
```

Comment:
If it's too much work to restructure those macros to support trylock semantics, that's fine -- we can merge this and do that work separately.  I'm a little wary of repeating the "op" 3 times in `WT_WITH_LOCK`.